File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Tables/Session.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Tables
{
    [Table("Session")]
    public class Session : Entity
    {
        public Guid SessionId { get; set; } = Guid.Empty;
        public Guid UserId { get; set; } = Guid.Empty;
        public string RefreshToken { get; set; } = string.Empty;
        public DateTime Expired { get; set; } = DateTime.Now;
        public DateTime Time { get; set; } = DateTime.Now;
        public string? Place { get; set; }
        public string? Device { get; set; }

        protected override Session Init()
        {
            return new Session();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/entities/session.js
import Entity from "./entity";

class Session extends Entity {
    constructor() {
        super();
        this.SessionId = null;
        this.UserId = null
        this.RefreshToken = null;
        this.Expired = null;
        this.Time = null;
        this.Place = null;
        this.Device = null;
    }

    init() {
        return new Session();
    }
}

export default Session;